pkg/build/dockerfile-generator_test.go

```diff
@@ -50,6 +50,90 @@ func TestPorter_buildDockerfile(t *testing.T) {
 	assert.Equal(t, wantlines, gotlines)
 }
 
+func TestPorter_buildDockerfile_alternateManifestLocation(t *testing.T) {
+	t.Parallel()
+
+	c := config.NewTestConfig(t)
+	tmpl := templates.NewTemplates()
+	configTpl, err := tmpl.GetManifest()
+	require.Nil(t, err)
+
+	manifestPath := "alternate/porter.yaml"
+	c.TestContext.AddTestFileContents(configTpl, manifestPath)
+
+	m, err := manifest.LoadManifestFrom(c.Context, manifestPath)
+	require.NoError(t, err, "could not load manifest")
+
+	// ignore mixins in the unit tests
+	m.Mixins = []manifest.MixinDeclaration{}
+
+	mp := mixin.NewTestMixinProvider()
+	g := NewDockerfileGenerator(c.Config, m, filepath.Join(c.Context.Getwd(), manifestPath), tmpl, mp)
+	gotlines, err := g.buildDockerfile()
+	require.NoError(t, err)
+
+	wantlines := []string{
+		"FROM debian:stretch-slim",
+		"",
+		"ARG BUNDLE_DIR",
+		"",
+		"RUN apt-get update && apt-get install -y ca-certificates",
+		"",
+		"",
+		"COPY . $BUNDLE_DIR",
+		"RUN rm $BUNDLE_DIR/alternate/porter.yaml",
+		"RUN rm -fr $BUNDLE_DIR/.cnab",
+		"COPY .cnab /cnab",
+		"WORKDIR $BUNDLE_DIR",
+		"CMD [\"/cnab/app/run\"]",
+	}
+	assert.Equal(t, wantlines, gotlines)
+}
+
+func TestPorter_buildDockerfile_separateManifestLocation(t *testing.T) {
+	t.Parallel()
+
+	c := config.NewTestConfig(t)
+	tmpl := templates.NewTemplates()
+	configTpl, err := tmpl.GetManifest()
+	require.Nil(t, err)
+
+	manifestPath := "separate/porter.yaml"
+	c.TestContext.AddTestFileContents(configTpl, manifestPath)
+
+	m, err := manifest.LoadManifestFrom(c.Context, manifestPath)
+	require.NoError(t, err, "could not load manifest")
+
+	// Now remove the local manifest to simulate it existing separate
+	// from the build context directory
+	c.TestContext.FileSystem.RemoveAll(manifestPath)
```

Comment:
I think we can change this test to more closely mimic what we are testing for: the manifest is not in the build context directory.

```go
manifestPath := "/tmp/porter.yaml"
// check but I think pwd for the test is in a different directory than /tmp, so this should be enough to trigger the condition?
```